packages/app-shell-shared/src/hooks/useFilteredConfig.ts

```diff
@@ -0,0 +1,445 @@
+    processServices(config.services);
+  }
+
+  return metadata;
+};
+
+export const useLoadAllConditionHooks = (
+  bundleList: string[],
+): { hooks: ((() => UseConditionResult) | null)[]; isLoading: boolean } => {
+  const [hooks, setHooks] = useState<((() => UseConditionResult) | null)[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  // Load all condition modules
+  useEffect(() => {
+    if (bundleList.length === 0) {
+      setHooks([]);
+      setIsLoading(false);
+      return;
+    }
+
+    let cancelled = false;
+    setIsLoading(true);
+
+    // Load all bundles in parallel
+    Promise.all(
+      bundleList.map((bundle) =>
+        import(/* @vite-ignore */ bundle)
+          .then((module) => module.default as () => UseConditionResult)
+          .catch((error) => {
+            console.error(`Failed to load condition hook ${bundle}:`, error);
+            return null;
+          }),
+      ),
+    ).then((loadedHooks) => {
+      if (!cancelled) {
+        setHooks(loadedHooks);
+        setIsLoading(false);
+      }
+    });
+
+    return () => {
+      cancelled = true;
+    };
+  }, [bundleList]);
+
+  return { hooks, isLoading };
+};
+
+/**
+ * Filter config based on condition results using element keys
+ */
+export const filterConfigWithResults = (
+  config: HvAppShellConfig,
+  metadata: ConditionMetadata[],
+  results: Map<number, boolean>,
+): HvAppShellConfig => {
+  // Group metadata by element key for efficient AND logic
+  const metadataByElement = new Map<string | number, ConditionMetadata[]>();
+
+  for (const meta of metadata) {
+    const group = metadataByElement.get(meta.elementKey) ?? [];
+    group.push(meta);
+    metadataByElement.set(meta.elementKey, group);
+  }
+
+  // Build element results map with AND logic: all conditions must be true
+  const elementResults = new Map<string | number, boolean>();
+
+  for (const [elementKey, metas] of metadataByElement) {
+    const allTrue = metas.every((meta) => results.get(meta.hookIndex) ?? false);
+    elementResults.set(elementKey, allTrue);
+  }
+
+  const filterViews = (
+    views: HvAppShellViewsConfig[],
+  ): HvAppShellViewsConfig[] => {
+    const result: HvAppShellViewsConfig[] = [];
+
+    for (const view of views) {
+      if (view.$key === undefined) {
+        result.push(view);
+        continue;
+      }
+
+      const shouldInclude = elementResults.get(view.$key) ?? true;
+
+      if (!shouldInclude) {
+        continue;
+      }
+
+      if (view.views) {
+        result.push({
+          ...view,
+          views: filterViews(view.views),
+        });
+      } else {
+        result.push(view);
+      }
+    }
+
+    return result;
+  };
+
+  const filterMenus = (
+    menus: HvAppShellMenuConfig[],
+  ): HvAppShellMenuConfig[] => {
+    const result: HvAppShellMenuConfig[] = [];
+
+    for (const menu of menus) {
+      if (menu.$key === undefined) {
+        result.push(menu);
+        continue;
+      }
```

Comment:
All menus should have `$key`, not just those with conditions. That's the only way to later be able to effectively use this for, for example, React's component `key` property.